docs/.vuepress/plugins/search-box/match-query.js

```diff
@@ -0,0 +1,51 @@
+
+import get from 'lodash/get'
+
+export default (query, page, additionalStr = null) => {
+  let domain = get(page, 'title', '')
+
+  if (get(page, 'frontmatter.tags')) {
+    domain += ` ${page.frontmatter.tags.join(' ')}`
+  }
+
+  if (additionalStr) {
+    domain += ` ${additionalStr}`
+  }
+
+  return matchTest(query, domain)
+}
+
+const matchTest = (query, domain) => {
+  const escapeRegExp = str => str.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&')
+
+  // eslint-disable-next-line no-control-regex
+  const nonASCIIRegExp = new RegExp('[^\x00-\x7F]')
+
+  const words = query
+    .split(/\s+/g)
+    .map(str => str.trim())
+    .filter(str => !!str)
+
+  if (!nonASCIIRegExp.test(query)) {
+    // if the query only has ASCII chars, treat as English
+    const hasTrailingSpace = query.endsWith(' ')
+    const searchRegex = new RegExp(
+      words
+      .map((word, index) => {
+        if (words.length === index + 1 && !hasTrailingSpace) {
+          // The last word - ok with the word being "startswith"-like
+          return `(?=.*\\b${escapeRegExp(word)})`
+        } else {
+          // Not the last word - expect the whole word exactly
+          return `(?=.*\\b${escapeRegExp(word)}\\b)`
+        }
+      })
+      .join('') + '.+',
+      'gi'
+    )
```

Comment:
The fuzzy search in HOT returns a page as a result iff it contains at least one of the query strings (not necessarily as a whole word). It is enabled only for 3 pages of the docs. Otherwise there would be too many results.
```
words.some(word => domain.toLowerCase().indexOf(word) > -1);
```

I did something similar in HF. My method returns a page as a result iff it contains **all** the query strings in any order (not necessarily as whole words). I enabled it for all the pages in the docs. 
```
words.every(word => domain.toLowerCase().indexOf(word) > -1);
```

WDYT?

Reply:
I think it is good, thanks!